src/replicache.ts

```diff
@@ -358,10 +359,11 @@ export class Replicache implements ReadTransaction {
 
   // Callback for when window.onstorage fires which happens when a different tab
   // changes the db.
-  private _onStorage = (e: StorageEvent): void => {
-    if (e.key === storageKeyName(this._name)) {
-      this._checkChange(e.newValue as string);
-    }
+  private async _onStorage() {
+    // Cannot just use the value from the other tab, because it can be behind us.
+    // Also, in the case of memstore, it will have a totally different, unrelated
+    // hash chain.
+    this._checkChange(await this._getRoot());
```

Comment:
It's tempting to simply always re-read `this._getRoot()`, but I think Erik is correct to skip that in cases like a transaction just landing where we literally just received that info from the previous RPC.